src/RewindStreamException.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Http\Client\Common\Plugin;
```

Comment:
given how few classes we have i think its ok to have the exception in the main namespace. though the result when using this package together client-common will be that there is a lot of classes in this namespace: https://github.com/php-http/client-common/tree/master/src/Plugin

if we ever add exceptions for specific plugins in client-common we are likely to place them in an `Exception` sub namespace. for that reason i would prefer to also put this exception into a sub namespace Http\Client\Common\Plugin\Exception.